### PR TITLE
Bad bug fixes

### PR DESCRIPTION
**Pull request by josh13w**
Fix bugs found in full stack testing: 

- New users not able to view samples when first joining organization 
- When a user deletes their account their document in the "users" collection was not deleted
- An org email was not created when a user creates a new org 

Manual testing: 
- Creating a new account, being approved to an organization and making sure the new account can view samples from that organization
- Deleting that new account and confirming their document is deleted from the "users" collection
- Creating a new organization and making sure an org_email is added to the orgs document 

Tests run: `npm run test`
```
Test Suites: 8 passed, 8 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        1.704 s, estimated 2 s
Ran all test suites.
```

### CODE REVIEW

File: sample_tracking/app/samples/page.tsx [hunk 1]
```diff
@@ -37,11 +37,18 @@ export default function Samples() {
         showTopBar();
         if (!userData.role || userData.role.length < 1) {
             onAuthStateChanged(auth, (user) => {
+                console.log("got here")
```
Comment (benwulfe) [inline sample_tracking/app/samples/page.tsx:40]: turd line, pls remove

File: sample_tracking/app/samples/page.tsx [hunk 2]
```diff
@@ -37,11 +37,18 @@ export default function Samples() {
...
                 if (!user) {
                     router.push('/login');
                 } else {
                     getUserData(user.uid).then((userData: UserData) => {
-                        setUserData(userData);
+                        if (userData.org) {
+                            console.log("got here!!! 1")
```
Comment (benwulfe) [inline sample_tracking/app/samples/page.tsx:46]: turd

File: sample_tracking/app/samples/page.tsx [hunk 3]
```diff
@@ -37,11 +37,18 @@ export default function Samples() {
...
+                            setUserData(userData);
+                        } else {
+                            setUserData({name: "no_user"} as UserData)
```
Comment (benwulfe) [inline sample_tracking/app/samples/page.tsx:49]: why are you setting name here?

File: sample_tracking/app/sign-up-requests/page.tsx [hunk 4]
```diff
@@ -111,8 +111,10 @@ export default function SignUpRequests() {
         const dateString = `${date.getMonth() + 1} ${date.getDate()} ${date.getFullYear()}`;
         const orgId = getRanHex(20);
         const newOrgRef = doc(db, "organizations", orgId);
+        const orgEmail = `${orgName}@timberid.org`;
```
Comment (benwulfe) [inline sample_tracking/app/sign-up-requests/page.tsx:114]: i am a little confused as to how a org@timberid.org email can be useful to an organization.  Are we setting up a gmail account for the org that they can log into somewhere?

**Review by benwulfe — CHANGES_REQUESTED** [review]
some turd lines and some questions about the code.

File: sample_tracking/app/sign-up-requests/page.tsx (hunk 4 above)
Comment (josh13w) [reply sample_tracking/app/sign-up-requests/page.tsx:114]: This was requested by Kader for EE. I'm not exactly sure how he uses it but its required for things on his end.

Comment (benwulfe) [reply sample_tracking/app/sign-up-requests/page.tsx:114]: ahh ok -- he uses it for folder permissions in EE.  makes sense

**Review by benwulfe — APPROVED** [review]
lgtm

**Pull request merged**